src/directives/infinite-scroll/infinite-scroll.component.spec.ts

```diff
@@ -133,3 +133,67 @@ describe('Directive - Infinite Scroll', () => {
         expect(loadSpy).not.toHaveBeenCalled();
     }));
 });
+
+@Component({
+    template: `
+        <div
+            style="height: 200px; overflow: auto;"
+            [uxInfiniteScroll]="load"
+            [pageSize]="20"
+            [loadOnScroll]="true"
+        >
+            <!-- Increase scroll height -->
+            <div style="height: 220px;"></div>
+        </div>
+    `,
+})
+export class InfiniteScrollTestDelayComponent {
+    @ViewChild(InfiniteScrollDirective) infiniteScrollDirective: InfiniteScrollDirective;
+
+    load(pageNum: number): Promise<any[]> {
+        const items = [];
+        return new Promise(resolve => {
+            setTimeout(() => {
+                for (let idx = pageNum * 20; idx < (pageNum + 1) * 20; idx++) {
+                    items.push(`Item ${idx}`);
+                }
+                resolve(items);
+            }, 2000);
+        });
+    }
+}
+
+describe('Directive - Infinite Scroll Promise', () => {
+    let component: InfiniteScrollTestDelayComponent;
+    let fixture: ComponentFixture<InfiniteScrollTestDelayComponent>;
+    let loadSpy: jasmine.Spy;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [InfiniteScrollModule, FormsModule],
+            declarations: [InfiniteScrollTestDelayComponent],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(InfiniteScrollTestDelayComponent);
+        component = fixture.componentInstance;
+        loadSpy = spyOn(component, 'load').and.callThrough();
+        fixture.detectChanges();
+    });
+
+    // checking if promise resolves twice
+    it('should resolve the load function when reset is called during load', () => {
+        expect(loadSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should allow loading to be triggered after reset, regardless if there are any pending requests', fakeAsync(() => {
+        component.infiniteScrollDirective.reset(true);
+        component.infiniteScrollDirective.check();
```

Comment:
In the repro sandbox, they are only calling `reset()` without an argument and without calling `check`.
```
this.usersList.reset();
```
Could you add a test case which matches this scenario?